analyses/fusion-frequencies/01-fusion-frequencies.R

```diff
@@ -0,0 +1,242 @@
+                  "Fusion_anno"=""))%>%
+  # We want a single column that contains the gene symbols
+  tidyr::gather(Gene1A, Gene1B, Gene2A, Gene2B,
+                key = gene_position, value = gene_symbol) %>%
+  filter(!is.na(gene_symbol)) %>%
+  dplyr::rename(Gene_Position = gene_position,
+                Gene_Symbol = gene_symbol)
+
+print(alt_id)
+
+if (opt$alt_id == "FusionName,Fusion_Type") {
+  fusion_df <- fusion_df %>%
+    mutate( Alt_ID=paste(!!as.name(alt_id),sep="_"))
+  keep_cols <- c("FusionName","Fusion_Type","Gene_Symbol", "Gene_Position",
+                 "Fusion_anno","BreakpointLocation", "annots",
+                 "Kinase_domain_retained_Gene1A",
+                 "Kinase_domain_retained_Gene1B",
+                 "Reciprocal_exists_either_gene_kinase",
+                 "Gene1A_anno","Gene1B_anno","Gene2A_anno","Gene2B_anno")
+  
+} else if (opt$alt_id == "Gene_Symbol") {
+  fusion_df <- fusion_df %>%
+    mutate( Alt_ID = Gene_Symbol )
+  keep_cols <-c("Gene_Symbol")
+
+}
+
+rm(tumor_kfbids)
+
+
+# Compute mutation frequencies -------------------------------------------------
+message('Compute mutation frequencies...')
+cancer_group_cohort_summary_df <- get_cg_cs_tbl(td_htl_dfs$overall_htl_df)
+
+# nf = n_samples filtered
+nf_cancer_group_cohort_summary_df <- cancer_group_cohort_summary_df %>%
+  filter(n_samples >= 5)
+
+# Run get_cg_ch_mut_freq_tbl() per cancer_group and cohort
+fus_freq_tbl_list <- lapply(
+  1:nrow(nf_cancer_group_cohort_summary_df),
+  function(i) {
+    cgcs_row <- nf_cancer_group_cohort_summary_df[i, ]
+    stopifnot(identical(nrow(cgcs_row), as.integer(1)))
+    stopifnot(identical(length(cgcs_row$cohort_list), as.integer(1)))
+    stopifnot(is.character(cgcs_row$cohort_list[[1]]))
+    
+    c_cancer_group <- cgcs_row$cancer_group
+    c_cohorts <- cgcs_row$cohort_list[[1]]
+    stopifnot(identical(paste(c_cohorts, collapse = '&'), cgcs_row$cohort))
+    message(paste(c_cancer_group, cgcs_row$cohort))
+    
+    # Call function for each cohort and cancer_group
+    res_df <- get_cg_ch_mut_freq_tbl(
+      fusion_df, td_htl_dfs$overall_htl_df, td_htl_dfs$primary_htl_df,
+      td_htl_dfs$relapse_htl_df, c_cancer_group, c_cohorts) %>%
+      # if Alt_ID has no matched Patients_in_dataset
+      filter(!is.na(Patients_in_dataset))
+    
+    if ( nrow(res_df) != 0 ){
+      # merge fusion dataframe with the counts and frequencies in each cancer_group
+      # - per cohort 
+      # - per primary samples in cohort
+      # - per relapse samples in cohort
+      fusion_df <- fusion_df %>% 
+        select(-Kids_First_Biospecimen_ID,-Kids_First_Participant_ID) %>%
+        unique() %>%
+        inner_join(res_df,by="Alt_ID") 
+
+      return(fusion_df)
+    }
+  }
+)
+
+m_fus_freq_tbl <- bind_rows(fus_freq_tbl_list) %>%
+  distinct()
+
+m_fus_freq_tbl <- m_fus_freq_tbl %>%
+  mutate(Dataset = if_else(Dataset == 'PBTA&GMKF&TARGET',
+                           true = 'all_cohorts', false = Dataset))
```

Comment:
Currently, `all_cohorts` denotes not only `'PBTA&GMKF&TARGET'` but also any `Dataset` value that has more than one cohorts, e.g. `'PBTA&GMKF'` and `'GMKF&TARGET'`.

```suggestion
m_fus_freq_tbl <- m_fus_freq_tbl %>%
  mutate(Dataset = if_else(str_detect(Dataset, '&'),
                           true = 'all_cohorts', false = Dataset))
```